### ISSUE TITLE
GENERIC: foo ( -- ) should allow exactly one method

### ISSUE THREAD

**Issue by erg** [slavapestov/factor#1421]
```factor
! No error on this one
GENERIC: foo ( -- )
```

We don't really have the syntax to allow this case anyway right now.

As an aside, all words could be generic by default if we special-cased generics with only one method.

**Comment by mrjbq7** [comment]
This makes that an error and also makes ``GENERIC# foo 1000 ( x -- )`` an error.

What do you think? Also, right now I just throw ``bad-generic-effect`` with combination and effect... maybe a nicer error message would be better?

```diff
diff --git a/core/generic/generic.factor b/core/generic/generic.factor
index 88ab8ef..e445f68 100644
--- a/core/generic/generic.factor
+++ b/core/generic/generic.factor
@@ -2,8 +2,8 @@
 ! See http://factorcode.org/license.txt for BSD license.
 USING: accessors arrays assocs classes classes.algebra
 classes.algebra.private classes.maybe classes.private
-combinators definitions kernel make namespaces sequences sets
-words ;
+combinators definitions kernel make math namespaces sequences
+sets words ;
 IN: generic
 
 ! Method combination protocol
@@ -11,6 +11,12 @@ GENERIC: perform-combination ( word combination -- )
 
 GENERIC: make-default-method ( generic combination -- method )
 
+ERROR: bad-generic-effect combination effect ;
+
+GENERIC# check-generic-effect 1 ( combination effect -- n )
+
+M: object check-generic-effect nip ;
+
 PREDICATE: generic < word
     "combination" word-prop >boolean ;
 
@@ -187,7 +193,7 @@ M: method forget*
     ] if ;
 
 : define-generic ( word combination effect -- )
-    [ nip swap set-stack-effect ]
+    [ check-generic-effect swap set-stack-effect ]
     [
         drop
         2dup [ "combination" word-prop ] dip = [ 2drop ] [

diff --git a/core/generic/standard/standard.factor b/core/generic/standard/standard.factor
index e8b342e..5f64f51 100644
--- a/core/generic/standard/standard.factor
+++ b/core/generic/standard/standard.factor
@@ -13,6 +13,10 @@ TUPLE: standard-combination < single-combination # ;
     dup 0 < [ bad-dispatch-position ] when
     standard-combination boa ;
 
+M: standard-combination check-generic-effect
+    2dup [ dispatch# ] [ in>> length ] bi* <
+    [ nip ] [ bad-generic-effect ] if ;
+
 PREDICATE: standard-generic < generic
     "combination" word-prop standard-combination? ;
```

**Comment by bjourne** [comment]
You mean that it should be an error because the left side of the stack effect is empty, right? Because it has nothing to dispatch on.

**Comment by erg** [comment]
It could dispatch on "nothing" once, but we don't even have syntax for
this. So we can disallow it for now.

If that fixes the other bug, cool. I can't test ATM.

**Comment by mrjbq7** [comment]
Fixed in 4fbbe6b2014c3d39b3467085bc27d6e41140bf92.

**Comment by erg** [comment]
How about this:
```
GENERIC# foo 2/3 ( a b c -- )
```

Lol. Maybe ensure it's an integer too?

**Comment by mrjbq7** [comment]
Fixed in 9aab1c87b45dc820fa9407a1138fe74f42e0cfdc.

That be crazy yo.
